src/main/java/com/minecolonies/permissions/ColonyPermissionEventHandler.java

```diff
@@ -0,0 +1,232 @@
+package com.minecolonies.permissions;
+
+import com.minecolonies.blocks.AbstractBlockHut;
+import com.minecolonies.colony.Colony;
+import com.minecolonies.colony.permissions.Permissions;
+import net.minecraft.block.Block;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.event.entity.item.ItemTossEvent;
+import net.minecraftforge.event.entity.player.PlayerContainerEvent;
+import net.minecraftforge.event.entity.player.PlayerInteractEvent;
+import net.minecraftforge.event.world.BlockEvent;
+import net.minecraftforge.event.world.ExplosionEvent;
+import net.minecraftforge.fml.common.eventhandler.Event;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * This class handles all permission checks on events and cancels them if needed.
+ */
+public class ColonyPermissionEventHandler
+{
+
+    private final Colony colony;
+
+    /**
+     * Create this EventHandler.
+     *
+     * @param colony the colony to check on.
+     */
+    public ColonyPermissionEventHandler(final Colony colony)
+    {
+        this.colony = colony;
+    }
+
+    /**
+     * BlockEvent.PlaceEvent handler.
+     *
+     * @param event BlockEvent.PlaceEvent
+     */
+    @SubscribeEvent
+    public void on(final BlockEvent.PlaceEvent event)
+    {
+        if (checkBlockEventDenied(event.getWorld(), event.getPos(), event.getPlayer(), event.getPlacedBlock()))
+        {
+            cancelEvent(event);
+        }
+    }
+
+    private static void cancelEvent(final Event event)
+    {
+        event.setResult(Event.Result.DENY);
+        if (event.isCancelable())
+        {
+            event.setCanceled(true);
+        }
+    }
+
+    /**
+     * BlockEvent.BreakEvent handler.
+     *
+     * @param event BlockEvent.BreakEvent
+     */
+    @SubscribeEvent
+    public void on(final BlockEvent.BreakEvent event)
+    {
+        if (checkBlockEventDenied(event.getWorld(), event.getPos(), event.getPlayer(), event.getWorld().getBlockState(event.getPos())))
+        {
+            cancelEvent(event);
+        }
+    }
+
+    /**
+     * ExplosionEvent.Detonate handler.
+     *
+     * @param event ExplosionEvent.Detonate
+     */
+    @SubscribeEvent
+    public void on(final ExplosionEvent.Detonate event)
+    {
+
+        final World eventWorld = event.getWorld();
+        Predicate<BlockPos> getBlocksInColony = pos -> colony.isCoordInColony(eventWorld, pos);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'getBlocksInColony' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)